### PR TITLE
DSND-2167: Add new DELETE endpoint to filing_history-api

### PR DESCRIPTION
**Pull request by crowlandsCH**
## Describe the changes

Local testing complete

### Related Jira tickets

[DSND-2167](https://companieshouse.atlassian.net/browse/DSND-2167)

### General

- [x] Is the PR as small as it can be?
- [x] Has the Jira ticket been updated with any relevant comments?
- [x] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [x] Has the code been double-checked against `main`?
- [x] Does the code adhere standards in this repository, including SonarQube checks?

### Testing

- [x] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation

_Where possible, add links to the respective Jira tickets when an item is checked_

- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to deployment repositories?

## Notes to the tester

_Add any testing hints or instructions here._

[DSND-2167]: https://companieshouse.atlassian.net/browse/DSND-2167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryController.java [hunk 1]
```diff
@@ -64,4 +70,19 @@ public ResponseEntity<ExternalData> getSingleFilingHistory(
                 .status(HttpStatus.OK)
                 .body(filingHistoryGetResponseProcessor.processGetSingleFilingHistory(transactionId, companyNumber));
     }
+
+    @DeleteMapping("/filing-history-data-api/filing-history/{transaction_id}")
```
Comment (cgregoryCH) [inline src/main/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryController.java:74]: My bad, URI should be ```/filing-history-data-api/filing-history/{transaction_id}/internal```

File: src/test/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryControllerIT.java [hunk 2]
```diff
@@ -486,6 +518,43 @@ void shouldReturn503ServiceUnavailableWhenChsKafkaApiReturnsA503ResponseAndDocum
         verify(instantSupplier, times(2)).get();
         WireMock.verify(
                 requestMadeFor(new ResourceChangedRequestMatcher(RESOURCE_CHANGED_URI, getExpectedChangedResource())));
+
+    }
+
+    @Test
```
Comment (cgregoryCH) [inline src/test/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryControllerIT.java:524]: Would be good to add a test case for when the item to delete is not in the database, 404 not found.

File: src/main/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryController.java (hunk 1 above)
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryController.java:74]: done

File: src/test/java/uk/gov/companieshouse/filinghistory/api/repository/RepositoryTest.java [hunk 3]
```diff
@@ -98,4 +98,17 @@ void shouldCatchDataAccessExceptionAndThrowServiceUnavailableWhenDeleteById() {
...
+    void shouldCatchDataAccessExceptionAndThrowServiceUnavailableWhenFindById() {
+        // given
+        when(mongoTemplate.findOne(any(), eq(FilingHistoryDocument.class))).thenThrow(new DataAccessException("...") {
+        });
+
+        // when
+        Executable executable = () -> repository.findById(TRANSACTION_ID);
+
+        // then
+        assertThrows(ServiceUnavailableException.class, executable);
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/test/java/uk/gov/companieshouse/filinghistory/api/repository/RepositoryTest.java:112]: done

File: src/test/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryControllerIT.java (hunk 2 above)
Comment (crowlandsCH) [reply src/test/java/uk/gov/companieshouse/filinghistory/api/controller/FilingHistoryControllerIT.java:524]: done

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by cgregoryCH — APPROVED** [review]

**Review by kinpang-CH — APPROVED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Pull request merged**